expan/core/early_stopping.py

```diff
@@ -0,0 +1,228 @@
+import os
+from os.path import dirname, join, realpath
+import numpy as np
+from scipy.stats import gaussian_kde, norm, cauchy
+from pystan import StanModel
+
+import expan.core.statistics as statx
+
+__location__ = realpath(join(os.getcwd(), dirname(__file__)))
+
+def obrien_fleming(information_fraction, alpha=0.05):
+	"""
+	Calculate an approximation of the O'Brien-Fleming alpha spending function.
+
+	Args:
+		information_fraction: share of the information amount at the point 
+			of evaluation, e.g. the share of the maximum sample size
+		alpha: type-I error rate
+
+	Returns:
+		float: redistributed alpha value at the time point with the given 
+			   information fraction
+	"""
+	return (1-norm.cdf(norm.ppf(1-alpha/2)/np.sqrt(information_fraction)))*2
+
+
+def group_sequential(x, 
+					 y, 
+					 spending_function='obrien_fleming', 
+					 information_fraction=1, 
+					 alpha=0.05, 
+					 cap=8):
+	"""
+	Group sequential method to determine whether to stop early or not.
+
+	Args:
+		x (array_like): sample of a treatment group
+		y (array_like): sample of a control group
+		spending_function: name of the alpha spending function, currently
+			supports: 'obrien_fleming'
+		information_fraction: share of the information amount at the point 
+			of evaluation, e.g. the share of the maximum sample size
+		alpha: type-I error rate
+		cap: upper bound of the adapted z-score
+
+	Returns:
+		tuple: 
+			- stop label
+			- effect size (delta)
+			- confidence interval of delta
+			- sample size of x
+			- sample size of y
+			- absolute mean of x
+			- absolute mean of y
+	"""
+	# Checking if data was provided
+	if x is None or y is None:
+		raise ValueError('Please provide two non-None samples.')
+
+	# Coercing missing values to right format
+	_x = np.array(x, dtype=float) 
+	_y = np.array(y, dtype=float) 
+
+	# if scalar, assume equal spacing between the intervals
+	#if not isinstance(information_fraction, list): 
+	#	fraction = np.linspace(0,1,information_fraction+1)[1:]
+	#else:
+	#	fraction = information_fraction
+
+	# alpha spending function
+	if spending_function in ('obrien_fleming'):
```

Comment:
continuing the comment above about passing the function itself instead of its name:

```
if callable(spending_function):
  alpha_new = spending_function(information_fraction, alpha = alpha)
else:
  raise NotImplemented
```

`callable()` works only in Python 2.x and Python 3.2+ though